Hazel/src/Platform/OpenGL/OpenGLShader.cpp

```diff
@@ -73,18 +73,21 @@ namespace Hazel {
 
 		const char* typeToken = "#type";
 		size_t typeTokenLength = strlen(typeToken);
-		size_t pos = source.find(typeToken, 0);
+		size_t pos = source.find(typeToken, 0);                                             //Start of shader type declaration line
 		while (pos != std::string::npos)
 		{
-			size_t eol = source.find_first_of("\r\n", pos);
+			size_t eol = source.find_first_of("\r\n", pos);                                 //End of shader type declaration line
 			HZ_CORE_ASSERT(eol != std::string::npos, "Syntax error");
-			size_t begin = pos + typeTokenLength + 1;
+			size_t begin = pos + typeTokenLength + 1;                                       //Start of shader type name (after "#type " keyword)
 			std::string type = source.substr(begin, eol - begin);
 			HZ_CORE_ASSERT(ShaderTypeFromString(type), "Invalid shader type specified");
 
-			size_t nextLinePos = source.find_first_not_of("\r\n", eol);
-			pos = source.find(typeToken, nextLinePos);
-			shaderSources[ShaderTypeFromString(type)] = source.substr(nextLinePos, pos - (nextLinePos == std::string::npos ? source.size() - 1 : nextLinePos));
+			size_t nextLinePos = source.find_first_not_of("\r\n", eol);                     //Start of shader code after shader type declaration line
+			HZ_CORE_ASSERT(nextLinePos != std::string::npos, "Syntax error");
+			pos = source.find(typeToken, nextLinePos);                                      //Start of next shader type declaration line
+
+			if (pos == std::string::npos) shaderSources[ShaderTypeFromString(type)] = source.substr(nextLinePos);
+			else shaderSources[ShaderTypeFromString(type)] = source.substr(nextLinePos, pos - nextLinePos);
```

Comment:
Please, for readability, split these if/else up in 4 lines (you can still omit the `{}`), but seperate the statement and the actual line of code into two lines:
```cpp
			if (pos == std::string::npos)
				shaderSources[ShaderTypeFromString(type)] = source.substr(nextLinePos);
			else
				shaderSources[ShaderTypeFromString(type)] = source.substr(nextLinePos, pos - nextLinePos);
```
I don't see why you removed the ternary conditional operator? (maybe add some `()` if you feel like it)
```cpp
			shaderSources[ShaderTypeFromString(type)] = pos == std::string::npos ? source.substr(nextLinePos) : source.substr(nextLinePos, pos - nextLinePos);

Reply:
Not sure why I did this either. Probably missed this solution. I have changed it to use the ternary conditional operator again (with brackets).